python_scripts/ensemble_bagging.py

```diff
@@ -184,72 +181,156 @@ def bootstrap_sample(data, target):
+# a resutl compute `n` predicted values for the target varible. The final
+# prediction of the ensemble for the test data point is the average of those
+# `n` values.
+#
+# We can plot the averaged predictions from the previous example.
 
 # %%
 sns.scatterplot(x=data_train["Feature"], y=target_train, color="black",
                 alpha=0.5)
 
-target_predicted_forest = []
-for tree_idx, tree in enumerate(forest):
+bag_predictions = []
+for tree_idx, tree in enumerate(bag_of_trees):
     target_predicted = tree.predict(data_test)
     plt.plot(data_test, target_predicted, linestyle="--", alpha=0.8,
              label=f"Tree #{tree_idx} predictions")
-    target_predicted_forest.append(target_predicted)
+    bag_predictions.append(target_predicted)
 
-target_predicted_forest = np.mean(target_predicted_forest, axis=0)
-plt.plot(data_test, target_predicted_forest, label="Averaged predictions",
+bag_predictions = np.mean(bag_predictions, axis=0)
+plt.plot(data_test, bag_predictions, label="Averaged predictions",
          linestyle="-")
 plt.legend()
-plt.title("Predictions of individual and combined tree")
+_ = plt.title("Predictions of bagged trees")
 
 # %% [markdown]
+#
 # The unbroken red line shows the averaged predictions, which would be the
-# final preditions given by our 'bag' of decision tree regressors.
+# final preditions given by our 'bag' of decision tree regressors. Note that
+# the predictions of the ensemble is more stable because of the averaging
+# operation. As a result, the bag of trees as a whole is less likely to overfit
+# than the individual trees.
 #
 # ## Bagging in scikit-learn
 #
-# Scikit-learn implements bagging estimators. It takes a base model that is the
-# model trained on each bootstrap sample.
+# Scikit-learn implements the bagging procedure as a "meta-estimator", that is
+# an estimator that wraps another estimator: it takes a base model that is
+# cloned several times and trained independently on each bootstrap sample.
+#
+# The following shows how to build a bagging ensemble of decision trees. We set
+# `n_estimtators=100` instead of 3 in our manual implementation above to get
+# a stronger smoothing effect.
 
 # %%
 from sklearn.ensemble import BaggingRegressor
 
-bagging = BaggingRegressor(base_estimator=DecisionTreeRegressor(),
-                           n_estimators=3)
-bagging.fit(data_train, target_train)
-target_predicted_forest = bagging.predict(data_test)
+bagged_trees = BaggingRegressor(
+    base_estimator=DecisionTreeRegressor(max_depth=3),
+    n_estimators=100,
+)
+_ = bagged_trees.fit(data_train, target_train)
 
+# %% [markdown]
+#
+# Let us visualize the predictions of the ensemble on the same test data:
 # %%
 sns.scatterplot(x=data_train["Feature"], y=target_train, color="black",
                 alpha=0.5)
-plt.plot(data_test, target_predicted_forest, label="Bag of decision trees")
-plt.legend()
+plt.plot(data_test, bagged_trees.predict(data_test))
```

Comment:
Just for consistence with the other notebooks where we go a bit slower:

```suggestion
target_predicted = bagged_trees.predict(data_test)
plt.plot(data_test, target_predicted)
```

Reply:
+1. What I did not like was introducing the local variable in the previous cell and use it only once in a few lines after in the other cell.